### ISSUE TITLE
OSX-specific crash

### ISSUE THREAD

**Issue by angband-import** [angband/angband#3967]
**Reported by NickMcConnell on 18 Apr 2015 00:42 UTC**
A crash that occurs on OSX but not windows or linux - see http://angband.oook.cz/forum/showpost.php?p=100276&postcount=306

**Comment by angband-import** [comment]
**Comment by myshkin on 21 Apr 2015 06:50 UTC**
I believe that this is the same issue that I described [http://angband.oook.cz/forum/showpost.php?p=100811&postcount=61 here]. When the vampire dies, it drops objects, and drop_near() calls ignore_item_ok(), which eventually tries to access deleted object kinds.

```
Program received signal EXC_BAD_ACCESS, Could not access memory.
Reason: 13 at address: 0x0000000000000000
0x000000010006d2d7 in object_flavor_is_aware (obj=0x10986b4b8) at obj-identify.c:170
170		return obj->kind->aware;
(gdb) bt
#0  0x000000010006d2d7 in object_flavor_is_aware (obj=0x10986b4b8) at obj-identify.c:170
#1  0x0000000100071910 in object_is_ignored (obj=0x10986b4b8) at obj-ignore.c:545
#2  0x0000000100071a82 in ignore_item_ok (obj=0x10986b4b8) at obj-ignore.c:579
#3  0x000000010007cf5e in drop_near (c=0x1088633d8, j_ptr=0x10986b4b8, chance=0, y=47, x=143, verbose=true) at obj-pile.c:871
#4  0x000000010005bd08 in monster_death (mon=0x108600950, stats=false) at mon-make.c:1346
#5  0x000000010005c200 in mon_take_hit (mon=0x108600950, dam=65, fear=0x7fff5fbfe9af, note=0x0) at mon-make.c:1478
```

**Comment by angband-import** [comment]
**Comment by NickMcConnell on 22 Apr 2015 13:25 UTC**
Fixed in 9dbd803
